pandas/tools/merge.py

```diff
@@ -981,16 +981,30 @@ def get_result(self):
             if self.axis == 0:
                 new_data = com._concat_compat([x._values for x in self.objs])
                 name = com._consensus_name_attr(self.objs)
-                return (Series(new_data, index=self.new_axes[0],
-                               name=name,
-                               dtype=new_data.dtype)
+
+                if com.is_sparse(new_data):
```

Comment:
yes, see #12503 (could close that bullet point). We can do the move of the others in another issue.

(your point 3), though *maybe* move ``tools/merge.py`` itself. So that would involve a slightly bigger reorg. Trying to think of a nice clean way of reorging this.